xbmc/cores/VideoPlayer/DVDCodecs/Video/DVDVideoCodecStarfish.cpp

```diff
@@ -0,0 +1,543 @@
+  payloadArg["option"]["bufferControl"]["userBufferCtrl"] = true;
+  payloadArg["option"]["bufferControl"]["bufferingMinTime"] = 0;
+  payloadArg["option"]["bufferControl"]["bufferingMaxTime"] = 1000000000;*/
+
+  payloadArgs["args"] = {payloadArg};
+
+  payload = payloadArgs.dump();
+  CLog::Log(LOGDEBUG, "CDVDVideoCodecStarfish: Sending Load payload {}", payload);
+  if (!m_starfishMediaAPI->Load(payload.c_str(), &CDVDVideoCodecStarfish::PlayerCallback, this))
+  {
+    CLog::Log(LOGERROR, "CDVDVideoCodecStarfish: Load failed");
+    m_InstanceGuard.exchange(false);
+    return false;
+  }
+
+  SetHDR();
+
+  m_codecControlFlags = 0;
+
+  CLog::Log(LOGINFO,
+            "CDVDVideoCodecStarfish:: "
+            "Open Starfish {}",
+            m_codecname);
+
+  // first make sure all properties are reset.
+  m_videobuffer.Reset();
+
+  m_videobuffer.iWidth = m_hints.width;
+  m_videobuffer.iHeight = m_hints.height;
+  m_videobuffer.iDisplayWidth = m_hints.width;
+  m_videobuffer.iDisplayHeight = m_hints.height;
+  m_videobuffer.stereoMode = m_hints.stereo_mode;
+  m_videobuffer.videoBuffer = new CStarfishVideoBuffer(0);
+
+  m_opened = true;
+
+  m_processInfo.SetVideoDecoderName(m_formatname, true);
+  m_processInfo.SetVideoPixelFormat("Surface");
+  m_processInfo.SetVideoDimensions(m_hints.width, m_hints.height);
+  m_processInfo.SetVideoDeintMethod("hardware");
+  m_processInfo.SetVideoDAR(m_hints.aspect);
+
+  UpdateFpsDuration();
+
+  return true;
+}
+
+void CDVDVideoCodecStarfish::Dispose()
+{
+  if (!m_opened)
+    return;
+  m_opened = false;
+
+  m_starfishMediaAPI->Unload();
+
+  m_InstanceGuard.exchange(false);
+}
+
+bool CDVDVideoCodecStarfish::AddData(const DemuxPacket& packet)
+{
+
+  if (!m_opened)
+    return false;
+
+  double pts(packet.pts), dts(packet.dts);
+
+  if (CServiceBroker::GetLogging().CanLogComponent(LOGVIDEO))
+    CLog::Log(LOGDEBUG,
+              "CDVDVideoCodecStarfish::AddData dts:{:0.2f} pts:{:0.2f} sz:{} "
+              "current state ({})",
+              dts, pts, packet.iSize, m_state);
+
+  if (dts == DVD_NOPTS_VALUE)
+    dts = 0;
+
+  if (m_hints.ptsinvalid)
+    pts = DVD_NOPTS_VALUE;
+
+  uint8_t* pData(packet.pData);
+  size_t iSize(packet.iSize);
+
+  // we have an input buffer, fill it.
+  if (pData && m_bitstream)
+  {
+    m_bitstream->Convert(pData, iSize);
+
+    if (m_state == STARFISH_STATE_FLUSHED && !m_bitstream->CanStartDecode())
+    {
+      CLog::Log(LOGDEBUG, "CDVDVideoCodecStarfish::AddData: waiting for keyframe (bitstream)");
+      return true;
+    }
+
+    iSize = m_bitstream->GetConvertSize();
+    pData = m_bitstream->GetConvertBuffer();
+  }
+
+  if (m_state == STARFISH_STATE_FLUSHED)
+  {
+    if (pts > 0)
+      m_starfishMediaAPI->Seek(std::to_string(DVD_TIME_TO_MSEC(pts)).c_str());
+    m_state = STARFISH_STATE_RUNNING;
+  }
+
+  if (pData && iSize)
+  {
+    nlohmann::json payload;
+    payload["bufferAddr"] = fmt::format("{}", fmt::ptr(pData));
```

Comment:
does it have to be formatted, actually? could you just cast to uintptr_t here? or use std::to_string?
using fmt::format for pointer addresses seems odd

Reply:
The API expects the address in this format 0xffffffff

Reply:
then let's do fmt::format with {:#x} or similar. the problem with fmt::ptr is that formatting a pointer is not guaranteed to lead to a specific representation. It doesn't get more specific than "Pointer format" in the docs...

Reply:
I spent some time looking into formatting and the best I could come up was {:#x}, (uint32_t)ptr. There's an assertion that doesn't let you format non void* pointers. I've added {:p} to make it more clear that we want a address to a pointer but you are right the fmt documentation doesn't say anything about how such a pointer would be formatted

Reply:
`{:#x}, reinterpret_cast<const void*>ptr` let to an segfault btw

Reply:
shouldn't ` {:#x}, reinterpret_cast<std::uintptr_t>(ptr)` work?

Reply:
Yep, that works